extensions/interactions/CodeRepl/CodeClassifierPredictionService.js

```diff
@@ -0,0 +1,217 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Code classifier prediction functions.
+ *
+ * IMPORTANT NOTE: The prediction function uses the classifier data
+ * of trained model for inference. These functions must be changed if
+ * there are any changes in corresponding classifier training function on
+ * Oppia-ml.
+ */
+
+oppia.factory('CodeClassifierPredictionService', [
```

Comment:
Yes -- look at how the interaction stuff gets loaded; there are files in the extensions/interaction/{{InteractionID}} directory and they are named in quite a specific way. So no, don't include things directly in CodeRepl.html. You probably need to:

- add a bool attr to the interaction .py file definition that indicates whether the interaction supports a prediction service
- for interactions that do support a prediction service, add that service to the {{InteractionId}}.js file
- write a backend test to ensure that all interactions have that boolean defined in their .py file, and that all interactions with that boolean set to True have a service with the appropriate name defined in their JS file. You can just do string matching on the raw JS file, I think.